main.c

```diff
@@ -212,6 +215,26 @@ int main(int argc, char *argv[])
         INFO("kibosh_main: configured %s.\n", conf_str);
     }
 
+    /* Reset random seed for the process. */
+    if (conf->random_seed) {
+        srand(conf->random_seed);
+        INFO("kibosh_main: random seed is set to %d.\n", conf->random_seed);
+    } else {
+        int s = (int) round(time(0));
+        srand(s);
+        INFO("kibosh_main: random seed is set to %d.\n", s);
+    }
+
+    /*
+     * Start a process to clear page cache every 1 second.
+     * The clear page cache process becomes necessary after direct_io option is removed.
+     * It serves two purposes:
+     * 1. preventing page cache from building up
+     * 2. force read and write calls to access the filesystem instead of the page cache to ensure fault can be injected
+     */
+    int sret = system("while true; do sleep 1; sudo sh -c \"echo 1 > /proc/sys/vm/drop_caches\"; done &");
```

Comment:
If this is really necessary then let's create a thread to do it, rather than using bash.

Reply:
Changed to child process as suggested.